code/processes/tailreader.q

```diff
@@ -1,32 +1,31 @@
 \d .tr
 partitiontype:@[value;`partitiontype;`date];                               /-set type of partition (defaults to `date)
 gmttime:@[value;`gmttime;1b];                                              /-define whether the process is on gmttime or not
-determinepartition:{@[value;`.tr.currentpartition;                         /-function to determine the partition value
-  (`date^partitiontype)$(.z.D,.z.d)gmttime]
+getpartition:{@[value;`getpartition;                       /-function to determine the partition value
+  {(`date^partitiontype)$(.z.D,.z.d)gmttime]}
   };
-getpartition:@[value;`getpartition;determinepartition[]];                  /-check if partition value exists and if not generate one
+currentpartition:@[value;`currentpartition;getpartition[]]
 segmentid: "J"$.proc.params[`segid]
 taildir:hsym `$getenv`KDBTAIL                                              /-load in taildir env variables
-currentpartition:getpartition;                                             /-obtain  partition value
 basedir:(raze/)1_string[.tr.taildir],"/tailer",string .tr.segmentid,"/"    /-define associated tailer base directory
-wdbdir:`$ basedir,string currentpartition                                  /-define IDB direction
+taildir:`$ basedir,string currentpartition                                  /-define tailDB direction
```

Comment:
This taildir overwrites the variable from line 9, so seems some of the code in lines 9 to 11 is redundant. If taildir from line 9 is no longer used after line 10 could probably combine these or something similar